src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,16 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        CommandResult result = null;
+
         try {
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageFile.StorageReadOnlyException e) {
+            ui.showToUser(e.getMessage());
```

Comment:
In general, just showing the error message or printing the stack trace is not considered "good exception handling".
Think how you can improve the user experience in light of the exception that just occurred.

Reply:
Thanks, I write this mainly because `e.getMessage()` will shows "Error writing to file" and "File is read only". However, I believe saving to a temporary file will be a better solution for this particular exception handling.